source/Calamari/Kubernetes/ResourceStatus/Resources/ResourceFactory.cs

```diff
@@ -16,38 +34,8 @@ public static IEnumerable<Resource> FromListJson(string json, Options options)
         
         public static Resource FromJObject(JObject data, Options options)
         {
-            var kind = data.SelectToken("$.kind")?.Value<string>();
-            switch (kind)
-            {   
-                case "Pod": 
-                    return new Pod(data, options);
-                case "ReplicaSet": 
-                    return new ReplicaSet(data, options);
-                case "Deployment":
-                    return new Deployment(data, options);
-                case "StatefulSet":
-                    return new StatefulSet(data, options);
-                case "DaemonSet":
-                    return new DaemonSet(data, options);
-                case "Job":
-                    return new Job(data, options);
-                case "CronJob":
-                    return new CronJob(data, options);
-                case "Service": 
-                    return new Service(data, options);
-                case "Ingress":
-                    return new Ingress(data, options);
-                case "EndpointSlice": 
-                    return new EndpointSlice(data, options); 
-                case "ConfigMap":
-                    return new ConfigMap(data, options);
-                case "Secret":
-                    return new Secret(data, options);
-                case "PersistentVolumeClaim":
-                    return new PersistentVolumeClaim(data, options);
-                default:
-                    return new Resource(data, options);
-            }
+            var gvk = data.ToResourceGroupVersionKind();
+            return resourceFactories.ContainsKey(gvk) ? resourceFactories[gvk](data, options) : new Resource(data, options);
```

Comment:
```suggestion
            return resourceFactories.TryGetValue(gvk, out var factory) ? factory(data, options) : new Resource(data, options);
```
1 less hash check in the dictionary